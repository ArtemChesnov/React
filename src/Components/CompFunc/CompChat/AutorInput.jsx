import React from 'react';

export const AutorInput = (props, { func }) => {
  return (
    <input
      value={props.autorValue}
      onChange={props.changeName}
      type="text"
      className="autor-input"
      required
      placeholder="Введите ваше имя..."
      click={func}
    />
  );
};
