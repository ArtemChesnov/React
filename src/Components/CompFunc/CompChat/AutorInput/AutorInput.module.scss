.autor__input {
    background: rgba(255, 255, 255, 0.15);
    outline: none;
    color: rgba(255, 255, 255, 0.9);
    outline: none;
    border: none;
    font-weight: 300;
    padding: 2px 0 2px 10px;
    height: 30px;
    font-family: BlinkMacSystemFont,
        "Segoe UI",
        "Roboto",
        "Oxygen",
        "Ubuntu",
        "Cantarell",
        "Fira Sans",
        "Droid Sans",
        "Helvetica Neue",
        sans-serif;
    -webkit-font-smoothing: antialiased;

    &::placeholder {
        font-size: 0.9em;
        color: rgba(255, 255, 255, 0.9);
        font-weight: 300;
    }
}