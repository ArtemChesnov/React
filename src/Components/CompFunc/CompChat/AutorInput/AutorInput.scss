.autor-input {
    width: 100%;
    resize: vertical;
    padding: 15px;
    border-radius: 15px;
    border: 0;
    box-shadow: 4px 4px 10px rgba(0, 0, 0, 0.06);
    margin: 0 0 25px 0;
}

.autor-input::placeholder {
    font-family: 'Trebuchet MS',
        'Lucida Sans Unicode',
        'Lucida Grande',
        'Lucida Sans',
        Arial,
        sans-serif;
}
