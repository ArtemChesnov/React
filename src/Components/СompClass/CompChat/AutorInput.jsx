import React, { Component } from 'react';

export class AutorInput extends Component {
  render() {
    return (
      <>
        <input
          value={this.props.autorValue}
          onChange={this.props.changeName}
          type="text"
          className="autor-input"
          placeholder="Введите ваше имя..."
          required
        />
      </>
    );
  }
}
