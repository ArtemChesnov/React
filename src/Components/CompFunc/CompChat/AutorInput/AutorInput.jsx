import React from 'react';
import './AutorInput.css';

export const AutorInput = ({ autorValue, setAutorValue }) => {
  return (
    <input
      value={autorValue}
      onChange={(event) => {
        setAutorValue(event.target.value);
      }}
      type="text"
      className="autor-input"
      required
      placeholder="Введите ваше имя..."
    />
  );
};
