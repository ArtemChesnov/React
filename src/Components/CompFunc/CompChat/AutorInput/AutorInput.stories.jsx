import React from 'react';
import { AutorInput } from './AutorInput';

export default {
  title: 'Components/AutorInput',
  component: AutorInput,
};

const Template = (args) => <AutorInput {...args} />;

export const Primary = Template.bind({});
