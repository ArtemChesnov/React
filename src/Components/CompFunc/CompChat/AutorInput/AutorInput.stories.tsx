import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { AutorInput } from './AutorInput';

export default {
  title: 'Components/AutorInput',
  component: AutorInput,
} as ComponentMeta<typeof AutorInput>;

const Template: ComponentStory<typeof AutorInput> = (args) => (
  <AutorInput {...args} />
);

export const Primary = Template.bind({});
