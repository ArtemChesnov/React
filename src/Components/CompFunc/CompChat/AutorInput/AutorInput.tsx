import React, { FC } from 'react';
import './AutorInput.scss';

interface AutorInputProps {
  autorValue: string;
  setAutorValue: (event: string) => void;
}

export const AutorInput: FC<AutorInputProps> = ({
  autorValue,
  setAutorValue,
}) => {
  return (
    <input
      value={autorValue}
      onChange={(event) => {
        setAutorValue(event.target.value);
      }}
      type="text"
      className="autor-input"
      required
      placeholder="Введите ваше имя..."
    />
  );
};
