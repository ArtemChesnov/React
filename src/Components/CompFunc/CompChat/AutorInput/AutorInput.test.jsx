import React from 'react';
import { AutorInput } from '../AutorInput/AutorInput';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('AutorInput', () => {
  it('render component', () => {
    render(<AutorInput />);
  });

  it('input to be in document', () => {
    render(<AutorInput />);
    expect(
      screen.getByPlaceholderText('Введите ваше имя...')
    ).toBeInTheDocument();
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<AutorInput />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('multiple render', () => {
    render(
      <>
        <AutorInput />
        <AutorInput />
      </>
    );
    expect(screen.getAllByRole('textbox').length).toBe(2);
  });

  it('render with value', () => {
    const mock = jest.fn();
    render(<AutorInput setAutorValue={mock} />);

    fireEvent.input(screen.getByPlaceholderText('Введите ваше имя...'), {
      target: { value: 'Name' },
    });
    expect(screen.getByDisplayValue(/Name/)).toBeInTheDocument();
  });

  it('input value change by typing', () => {
    const mock = jest.fn();
    render(<AutorInput setAutorValue={mock} />);

    const input = screen.getByPlaceholderText('Введите ваше имя...');

    fireEvent.change(input, { target: { value: 'Name' } });

    expect(input.value).toBe('Name');
  });
});
