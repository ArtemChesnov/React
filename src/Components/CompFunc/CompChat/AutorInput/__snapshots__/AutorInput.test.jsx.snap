// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutorInput render with snapshot 1`] = `
<DocumentFragment>
  <input
    class="autor-input"
    placeholder="Введите ваше имя..."
    required=""
    type="text"
    value=""
  />
</DocumentFragment>
`;
